school_management/tests/test_school_student.py

```diff
@@ -0,0 +1,78 @@
+from datetime import date
+
+from odoo.exceptions import ValidationError
+from odoo.tests.common import TransactionCase
+
+
+class TestSchoolStudent(TransactionCase):
+    def setUp(self):
+        super(TestSchoolStudent, self).setUp()
+        self.student_record = self.env["school.student"].create(
+            {
+                "student_name": "Manan",
+                "student_date_of_birth": date.today(),
+            }
+        )
+
+        self.school_ref = self.env["school.reference"].create(
+            {
+                "reference_name": "ron",
+                "reference_phone": "9968967890",
+                "discount_amount": "30%",
+            }
+        )
+        self.subject_record = self.env["school.subject"].create(
+            {"subject_name": "hindi", "subject_credit": 6}
+        )
+
+    def test_01_duplicate(self):
+        """checking if the inherited copy() method is working as intended #T00476"""
+        self.student_record.copy()
+
+    def test_02_reference_validation(self):
+        """validating weather the promo_validation method works as intended #T00476"""
+        self.student_record.write({"reference_code": self.school_ref.reference_promo})
+        self.student_record.promo_validation()
+        self.assertEqual(
+            self.school_ref.reference_name,
+            self.student_record.referral_name,
+            "the refrence name logic not performing as expected",
```

Comment:
The Test message should be "The school reference name does not match with the student referral name! "